DirectProgramming/DPC++FPGA/ReferenceDesigns/cholesky/src/cholesky_demo.cpp

```diff
@@ -150,31 +154,31 @@ int main(int argc, char *argv[]) {
 
       for (size_t row = 0; row < kRows; row++) {
         for (size_t col = 0; col < kColumns; col++) {
-          float diag_scaling = row == col ? float{kRows} : 0;
+          float diag_scaling = (row == col) ? float{kRows} : 0;
 
           if (col >= row) {
             float random_real = RandomValueInInterval(kRandomMin, kRandomMax);
 
 #if COMPLEX == 0
-            a_matrix[matrix_index * kAMatrixSize + col * kRows + row] =
+            a_matrix[(matrix_index * kAMatrixSize) + (col * kRows) + row] =
```

Comment:
Many of these matrix index calculations are repeated - please factor out, unless that impacts QoR.

If such factoring does impact QoR then consider logging a compiler bug.